xtra-libp2p-offer/src/current/maker.rs

```diff
@@ -52,14 +52,14 @@ impl Actor {
         };
 
         let err_handler = move |e: anyhow::Error| async move {
-            if let Some(xtra_libp2p::Error::NegotiationFailed(
-                xtra_libp2p::NegotiationError::Failed,
-            )) = e.downcast_ref::<xtra_libp2p::Error>()
-            {
-                // It's normal to disagree on the protocols now that we broadcast on both versions
-                // to _all_ our peers
-            } else {
-                tracing::warn!(%peer_id, "Failed to send offers: {e:#}")
+            match e.downcast_ref::<xtra_libp2p::Error>() {
+                Some(xtra_libp2p::Error::ProtocolNotSupportedByPeer) => {
+                    // Some peers may not support this protocol as listeners
```

Comment:
🤔 shouldn't we add a log message here?

Reply:
I don't think logging will add much other than noise.

The current expected behaviour is that we will attempt to open a substream but be stopped by the `Endpoint` if it knows that the peer doesn't listen. It's not noteworthy because these `offer` actors learn about all connected peers from the `Endpoint`, so we expect some of those peers to be incompatible.

What we could do is remove the incompatible peer from the internal state of the `offer::Actor`, so that we don't keep spamming the `Endpoint` if we know it's not gonna work.

Reply:
that sounds reasonable.. 🙂

Reply:
But let's add the extra functionality with #2839, because we already introduce an internal message for the same purpose with 0d1dd1bdb5cc12e38bfcf51dfc4d9cd43a58adf8.